### PR TITLE
[BugFix] Fix TP support for AWQ

### PR DESCRIPTION
**Pull request by WoosukKwon**
Tested:

- [x] LLaMA (`TheBloke/Llama-2-7B-AWQ`)
- [x] Mistral (`TheBloke/Mistral-7B-Instruct-v0.1-AWQ`)
- [x] OPT (`ybelkada/opt-125m-awq`)
- [ ] ~~Falcon~~ (No AWQ + Falcon-40B model available in HF hub) 
- [x] MPT (`TitanML/mpt-7b-chat-8k-4bit-AWQ`)
- [x] GPTBigCode (`casperhansen/starcoderbase-1b-awq`)
- [x] GPT-J (`TitanML/gptj-6b-base-4bit-AWQ`)
- [x] BLOOM (`bdambrosio/bloom-awq`)

### CODE REVIEW

**Review by zhuohan123 — COMMENTED** [review]
Should we add parallelization to the `ScaledActivation` class instead? And add weight loading functionalities for the `scales` tensor below.

https://github.com/vllm-project/vllm/blob/f5a37c6c6c8204ec42187525ebdb8288cfc4b552/vllm/model_executor/layers/activation.py#L45-L63

**Comment by WoosukKwon** [comment]
@zhuohan123 You're absolutely right. That's a far better design. Let me change the PR accordingly.

**Comment by WoosukKwon** [comment]
@zhuohan123 The PR is ready for review. PTAL.

File: vllm/model_executor/layers/activation.py [hunk 1]
```diff
@@ -51,17 +55,31 @@ class ScaledActivation(nn.Module):
     def __init__(
         self,
         act_module: nn.Module,
-        hidden_size: int,
+        intermediate_size: int,
         params_dtype: torch.dtype,
```
Comment (zhuohan123) [inline vllm/model_executor/layers/activation.py:59]: Maybe we can add an option here indicating whether the input is parallelized or not?

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks for the quick fix!

File: vllm/model_executor/layers/activation.py (hunk 1 above)
Comment (WoosukKwon) [reply vllm/model_executor/layers/activation.py:59]: Oh then what should we do if `input_is_parallel=False`?

Comment (WoosukKwon) [reply vllm/model_executor/layers/activation.py:59]: Got ya. fixed.

**Comment by YQ-Wang** [comment]
Does this PR provide enhancements for latency and throughput optimization in the AWQ models' inference process?

**Pull request merged**